srcpkgs/Aegisub-wangqr/template

```diff
@@ -0,0 +1,61 @@
+# Template file for 'Aegisub-wangqr'
+pkgname=Aegisub-wangqr
+version=3.3.2
+revision=1
+wrksrc="${pkgname}-${version}"
+build_style=cmake
+cmake_builddir="BUILD"
+configure_args="
+	-DwxWidgets_CONFIG_EXECUTABLE=/usr/bin/wx-config-gtk3
+	-DWITH_PORTAUDIO=$(vopt_if portaudio ON OFF)
+	-DWITH_OPENAL=$(vopt_if openal ON OFF)
+	-DWITH_FFTW3=$(vopt_if fftw ON OFF)
+"
+hostmakedepends="automake gettext-devel intltool libtool LuaJIT pkg-config git"
+makedepends="
+	alsa-lib-devel
+	boost-devel
+	libcurl-devel
+	fontconfig-devel
+	freetype-devel
+	hunspell-devel
+	icu-devel
+	libass-devel
+	libffms2-devel
+	LuaJIT-devel
+	MesaLib-devel
+	wxWidgets-gtk3-devel
+	$(vopt_if fftw fftw-devel)
+	$(vopt_if openal libopenal-devel)
+	$(vopt_if portaudio portaudio-devel)
+"
+short_desc="Tool for creating and modifying subtitles (wangqr fork)"
+maintainer="pudiva <rogi@skylittlesystem.org>"
+license="BSD-3-Clause, MIT"
+homepage="https://github.com/wangqr/Aegisub"
+conflicts=Aegisub
+
+build_options="fftw openal portaudio"
+build_options_default="fftw"
+
+desc_option_fftw="Enable support for rendering of audio waveforms/spectrum"
+desc_option_openal="Enable support for openal"
+
+do_fetch() {
+	git clone git://github.com/wangqr/Aegisub.git $wrksrc
+	cd $wrksrc
+	git checkout v$version
+}
+
```

Comment:
It would be much better to download the release tarball and patch the version detection script. I don't think it will be accepted this way.

Reply:
We have packages that do this, for a myriad reasons, but yes, I'd rather not *move* to such a setup.

At worst, `echo $version > somefile` and patch cmake to read from that instead.

Reply:
> At worst, echo $version > somefile and patch cmake to read from that instead.

Unfortunately this is how **build/version.sh** generates a new **git_version.h** file:

```sh
new_version_h="\
#define BUILD_GIT_VERSION_NUMBER ${git_revision}
#define BUILD_GIT_VERSION_STRING \"${git_version_str}\"
#define TAGGED_RELEASE ${tagged_release}
#define INSTALLER_VERSION \"${installer_version}\"
#define RESOURCE_BASE_VERSION ${resource_version}"
```

I'm concerned with the `BUILD_GIT_VERSION_NUMBER` macro, that is set to `$git_revision`, that is defined as:

```sh
last_svn_revision=6962
last_svn_hash="16cd907fe7482cb54a7374cd28b8501f138116be"

git_revision=$(expr $last_svn_revision + $(git log --pretty=oneline $last_svn_hash..HEAD 2>/dev/null | wc -l))
```

which basically counts the number of commits since  `$last_svn_hash` and adds `$last_svn_revision`.

Now there are three options:

1. fetch with git
2. generate **git_version.h** on `pre_configure()`
   2.1. using hardcoded values manually extracted from git by the pkg maintainer (me???) on every update
   2.2. using the template's `$version` and never updating the other values

I did a little grepping but gave up following where the variable is ultimately used, so I'm not sure how important it is :woman_shrugging: 

To be honest, I'd rather go with what the maintainer supports, which is building from git, instead of creating more pkg maintenance overhead trying to build from the release tarballs that they'd have removed if they could:

> There are no release tarballs. “Source Code (zip)” and “Source Code (tar.gz)” listed on the release page are just automatically generated archives of the source, _without the `.git`-folder_ thus making them useless for regular building.Afaik GitHub doesn't allow to disable those.

https://github.com/wangqr/Aegisub/issues/116#issuecomment-886105613

... and I'm not even sure this is the only problem with the tarballs.

What do you think? :sparkles: :sparkles: :sparkles:

Reply:
Point against building from git is that we lose caching, so every build has to repeat the huge download. The reason we do it for `ardour` is because upstream deletes their pre-generated tarballs, but I still greatly dislike it...

IMO let's move the PR forward by pulling from git, then ponder more from there.

Please do like it's done in `ardour`, where we checkout a specific commit.

Reply:
> Please do like it's done in ardour, where we checkout a specific commit.

You mean you want me to reference the commit hash instead of tag v3.3.2?

Reply:
> > Please do like it's done in ardour, where we checkout a specific commit.
> 
> You mean you want me to reference the commit hash instead of tag v3.3.2?

I did that already. Please, take a look and tell me what you think. :sparkles: